pkg/karmadactl/token.go

```diff
@@ -0,0 +1,725 @@
+package karmadactl
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	clientcertutil "k8s.io/client-go/util/cert"
+	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
+	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+	bootstrapsecretutil "k8s.io/cluster-bootstrap/util/secrets"
+	"k8s.io/klog/v2"
+	"k8s.io/kubectl/pkg/cmd/get"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/options"
+	"github.com/karmada-io/karmada/pkg/karmadactl/util/pubkeypin"
+)
+
+const (
+	// DefaultTokenDuration specifies the default amount of time that a bootstrap token will be valid
+	// Default behaviour is 24 hours
+	DefaultTokenDuration = 24 * time.Hour
+)
+
+var (
+	tokenLong = templates.LongDesc(`
+	This command manages bootstrap tokens. It is optional and needed only for advanced use cases.
+
+	In short, bootstrap tokens are used for establishing bidirectional trust between a client and a server.
+	A bootstrap token can be used when a client (for example a member cluster that is about to join control plane) needs
+	to trust the server it is talking to. Then a bootstrap token with the "signing" usage can be used.
+	bootstrap tokens can also function as a way to allow short-lived authentication to the API Server
+	(the token serves as a way for the API Server to trust the client), for example for doing the TLS Bootstrap.
+
+	What is a bootstrap token more exactly?
+	 - It is a Secret in the kube-system namespace of type "bootstrap.kubernetes.io/token".
```

Comment:
Can you confirm if the Secret will be created in `kube-system` namespace?

Reply:
Yes, the `tokencleaner` controller only delete expired tokens in `kube-system` namespace.